src/uu/head/src/head.rs

```diff
@@ -1,240 +1,557 @@
-//  * This file is part of the uutils coreutils package.
-//  *
-//  * (c) Alan Andrade <alan.andradec@gmail.com>
-//  *
-//  * For the full copyright and license information, please view the LICENSE
-//  * file that was distributed with this source code.
-//  *
-//  * Synced with: https://raw.github.com/avsm/src/master/usr.bin/head/head.c
-
-#[macro_use]
-extern crate uucore;
-
-use std::collections::VecDeque;
-use std::fs::File;
-use std::io::{stdin, BufRead, BufReader, Read};
-use std::path::Path;
-use std::str::from_utf8;
-
-static SYNTAX: &str = "";
-static SUMMARY: &str = "";
-static LONG_HELP: &str = "";
-
-enum FilterMode {
-    Bytes(usize),
+use clap::{App, Arg};
+use std::convert::TryFrom;
+use std::ffi::OsString;
+use std::io::{ErrorKind, Read, Seek, SeekFrom, Write};
+use uucore::{crash, executable, show_error};
+
+const EXIT_FAILURE: i32 = 1;
+const EXIT_SUCCESS: i32 = 0;
+const BUF_SIZE: usize = 65536;
+
+const VERSION: &str = env!("CARGO_PKG_VERSION");
+const ABOUT: &str = "\
+    Print the first 10 lines of each FILE to standard output.\n\
+    With more than one FILE, precede each with a header giving the file name.\n\
+    \n\
+    With no FILE, or when FILE is -, read standard input.\n\
+    \n\
+    Mandatory arguments to long flags are mandatory for short flags too.\
+    ";
+const USAGE: &str = "head [FLAG]... [FILE]...";
+
+mod options {
+    pub const BYTES_NAME: &str = "BYTES";
+    pub const BYTES_HELP: &str = "\
+        print the first NUM bytes of each file;\n\
+        with the leading '-', print all but the last\n\
+        NUM bytes of each file\
+        ";
+    pub const LINES_NAME: &str = "LINES";
+    pub const LINES_HELP: &str = "\
+        print the first NUM lines instead of the first 10;\n\
+        with the leading '-', print all but the last\n\
+        NUM lines of each file\
+        ";
+    pub const QUIET_NAME: &str = "QUIET";
+    pub const QUIET_HELP: &str = "never print headers giving file names";
+    pub const VERBOSE_NAME: &str = "VERBOSE";
+    pub const VERBOSE_HELP: &str = "always print headers giving file names";
+    pub const ZERO_NAME: &str = "ZERO";
+    pub const ZERO_HELP: &str = "line delimiter is NUL, not newline";
+    pub const FILES_NAME: &str = "FILE";
+}
+mod parse;
+mod split;
+
+fn app<'a>() -> App<'a, 'a> {
```

Comment:
please keep the same profile as others
https://github.com/uutils/coreutils/blob/master/src/uu/ls/src/ls.rs#L359
pub fn uumain(args: impl uucore::Args) -> i32 {

Reply:
I disagree with this change.

having an `app` function that constructs the clap `App` means we can unit test argument parsing. 
The alternative would be to have 2 copies of the same code, one for the unit tests, the other for the actual executable, which would be a nightmare to maintain, but also reduce readability.

Reply:
ok, no worries